doc/news/changes/incompatibilities/20170406MatthiasMaier

```diff
@@ -0,0 +1,6 @@
+Removed: A number of deprecated classes have been removed in favor of the
+new LinearOperator concept: TODO
```

Comment:
@bangerth This is the placeholder for the detailed description of every retired class and its replacement.